### PR TITLE
ACS-8482 Adapt the Agents API to recent changes

### PR DESCRIPTION
**Pull request by damianujma**
Changes related to return from `GET /agent`

### CODE REVIEW

File: e2e-test/src/test/resources/wiremock/hxinsight/mappings/get-agents.json [hunk 1]
```diff
@@ -7,27 +7,47 @@
...
-        "agentVersion": "1",
+        "version": "1",
         "description": "This agent is responsible for HR policy predictions",
-        "personality": "Keep responses professional and concise",
-        "model": "meta-llama3-8b",
-        "allowedSources": [
+        "instructions": "Keep responses professional and concise",
+        "modelName": "meta-llama3-8b",
+        "sourceIds": [
           "HR Policy Repository"
```
Comment (dsibilio) [inline e2e-test/src/test/resources/wiremock/hxinsight/mappings/get-agents.json:18]:
```suggestion
        "sourceIds": [
          "9036b44b-8811-4449-a767-3763e3efa2e0",
          "c2886904-40d0-4ab9-b5b7-1ea624ccca80"
```

File: distribution/src/main/resources/docker-compose/wiremock/hxinsight/__files/agents.json [hunk 2]
```diff
@@ -1,23 +1,45 @@
...
+    "version": "1",
     "description": "Very smart about product knowledge",
-    "personality": "Keep responses professional but friendly",
-    "model": "meta-llama3-8b",
-    "allowedSources": [
+    "instructions": "Keep responses professional but friendly",
+    "modelName": "meta-llama3-8b",
+    "sourceIds": [
       "Insight Discovery",
       "HR Policy Repository"
```
Comment (dsibilio) [inline distribution/src/main/resources/docker-compose/wiremock/hxinsight/__files/agents.json:33]:
```suggestion
    "sourceIds": [
      "9036b44b-8811-4449-a767-3763e3efa2e0",
      "c2886904-40d0-4ab9-b5b7-1ea624ccca80"
```

File: distribution/src/main/resources/docker-compose/wiremock/hxinsight/__files/agents.json [hunk 3]
```diff
@@ -1,23 +1,45 @@
...
+    ],
+    "permissions": [
+      {
+        "principalType": "user",
+        "id": "e2b9471c-d869-48b4-94ce-5112dd71001c"
+      }
+    ],
+    "avatarUrl": "https://s3.amazonaws.com/avatars/ecf13dd4-c061-462b-8122-e3203cc40a0d.jpg",
+    "staticFilter": {},
+    "dynamicFilter": {}
```
Comment (dsibilio) [inline distribution/src/main/resources/docker-compose/wiremock/hxinsight/__files/agents.json:21]: where do these come from? I didn't see them on the HX Swagger UI

File: distribution/src/main/resources/docker-compose/wiremock/hxinsight/__files/agents.json [hunk 4]
```diff
@@ -1,23 +1,45 @@
...
+    ],
+    "permissions": [
+      {
+        "principalType": "user",
+        "id": "5d684559-810b-499f-8b6b-4f39eb47d367"
+      }
+    ],
+    "avatarUrl": "https://s3.amazonaws.com/avatars/e2119fa1-9e11-421a-bef3-222667b1db34.jpg",
+    "staticFilter": {},
+    "dynamicFilter": {}
```
Comment (dsibilio) [inline distribution/src/main/resources/docker-compose/wiremock/hxinsight/__files/agents.json:43]: Same as above.

File: e2e-test/src/test/resources/wiremock/hxinsight/mappings/get-agents.json [hunk 5]
```diff
@@ -7,27 +7,47 @@
...
+        ],
+        "permissions": [
+          {
+            "principalType": "user",
+            "id": "e2b9471c-d869-48b4-94ce-5112dd71001c"
+          }
+        ],
+        "avatarUrl": "https://s3.amazonaws.com/avatars/ecf13dd4-c061-462b-8122-e3203cc40a0d.jpg",
+        "staticFilter": {},
+        "dynamicFilter": {}
```
Comment (dsibilio) [inline e2e-test/src/test/resources/wiremock/hxinsight/mappings/get-agents.json:28]: Same as above.

File: e2e-test/src/test/resources/wiremock/hxinsight/mappings/get-agents.json [hunk 6]
```diff
@@ -7,27 +7,47 @@
...
+        "version": "1",
         "description": "Very smart about product knowledge",
-        "personality": "Keep responses professional but friendly",
-        "model": "meta-llama3-8b",
-        "allowedSources": [
+        "instructions": "Keep responses professional but friendly",
+        "modelName": "meta-llama3-8b",
+        "sourceIds": [
           "Insight Discovery",
           "HR Policy Repository"
```
Comment (dsibilio) [inline e2e-test/src/test/resources/wiremock/hxinsight/mappings/get-agents.json:40]:
```suggestion
        "sourceIds": [
          "9036b44b-8811-4449-a767-3763e3efa2e0",
          "c2886904-40d0-4ab9-b5b7-1ea624ccca80"
```

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/model/Agent.java [hunk 7]
```diff
@@ -34,7 +34,7 @@
 @NoArgsConstructor
 public class Agent
 {
-    private String agentId;
+    private String id;
```
Comment (dsibilio) [inline hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/model/Agent.java:37]: I see quite a lot of changed fields but only agentId -> id has been changed here. I'm guessing we don't need to map any of them at the moment?

File: distribution/src/main/resources/docker-compose/wiremock/hxinsight/__files/agents.json (hunk 3 above)
Comment (damianujma) [reply distribution/src/main/resources/docker-compose/wiremock/hxinsight/__files/agents.json:21]: Please see in the `agentResponse` schema. 
See also: https://github.com/HylandSoftware/hxai-data-knowledge-retrieval/blob/539fc9efa9b66cd4fe2340745b5d8bfbdc05a64e/openapi/hxai-discovery-api.yaml

Comment (dsibilio) [reply distribution/src/main/resources/docker-compose/wiremock/hxinsight/__files/agents.json:21]: Can't access this link but I'll trust you on that.

@mpichura do you have access to the HylandSoftware repositories? In that case, what's the procedure we have to go through to also request access?

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/model/Agent.java (hunk 7 above)
Comment (damianujma) [reply hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/model/Agent.java:37]: For now we need only `id`, `name`, `description`.

**Review by Pawel-608 — APPROVED** [review]
LGTM

File: distribution/src/main/resources/docker-compose/wiremock/hxinsight/__files/agents.json (hunk 3 above)
Comment (krdabrowski) [reply distribution/src/main/resources/docker-compose/wiremock/hxinsight/__files/agents.json:21]: I think you need to be a part of https://github.com/orgs/HylandSoftware/teams/all-employees team. You can ask [Jakub Jeszke] (https://hyland.enterprise.slack.com/team/U039QEUJ6CU) about that

**Review by krdabrowski — APPROVED** [review]
LGTM

File: distribution/src/main/resources/docker-compose/wiremock/hxinsight/__files/agents.json (hunk 3 above)
Comment (mpichura) [reply distribution/src/main/resources/docker-compose/wiremock/hxinsight/__files/agents.json:21]: @dsibilio - sorry for late reply - I somehow missed that message. As Krystian mentioned - you can ask Jakub for access. There's also SwaggerUI endpoint available http://hxai-data-platform-dev-swagger-ui.s3-website-us-east-1.amazonaws.com/?urls.primaryName=Insight%20Discovery%20API%20Draft#/Discovery-Agent-API/list_agents, however, I am not sure how up to date that one is (those 2 fields are included there though).

**Review by mpichura — APPROVED** [review]
LGTM

**Pull request merged**